src/stactools/core/add.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+import pystac
+from pystac.layout import BestPracticesLayoutStrategy
+
+from stactools.core.copy import move_assets as do_move_assets
+
+
+def add_item(source_item, target_catalog, move_assets=False):
+    """Add a item into a catalog.
+
+    Args:
+        source_item (pystac.Item): The Item that will be added.
+            This item is not mutated in this operation.
+        target_catalog (pystac.Item): The destination catalog.
+            This catalog will be mutated in this operation.
+        move_assets (bool): If true, move the asset files alongside the target item.
+    """
+
+    target_item_ids = [item.id for item in target_catalog.get_all_items()]
+    if source_item.id in target_item_ids:
+        raise ValueError(
+            f'An item with ID {source_item.id} already exists in the target catalog'
+        )
+    parent_dir = os.path.dirname(target_catalog.get_self_href())
+    layout_strategy = BestPracticesLayoutStrategy()
+    item_copy = source_item.clone()
+    item_copy.set_self_href(
+        layout_strategy.get_item_href(item_copy, parent_dir))
+    target_catalog.add_item(item_copy)
+
+    if isinstance(target_catalog, pystac.Collection):
+        item_copy.set_collection(target_catalog)
```

Comment:
It makes sense to me, but not in this "add" command, I guess. Instead, better on the pystac part, anytime an item is added to a catalog, so it's done automatically not only when called from here

Reply:
> better on the pystac part, anytime an item is added to a catalog, so it's done automatically not only when called from here

I'm not sure if PySTAC should do this, in case you're performing `add_item` a bunch of times in a row and only want to do the extents calculation after the last add. After thinking about it, I think we _should_ update the collection's extents with the item -- otherwise, the collection's extents are in an ambiguous state, possibly correct but possibly incorrect.

Reply:
The extent of the collection is updated after the item is added. I have changed the syntax of that block, so it's clearer.